Ryujinx.Ava/UI/Windows/SettingsWindow.axaml.cs

```diff
@@ -55,159 +35,69 @@ public SettingsWindow()
             Load();
         }
 
-        private void Load()
+        public void SaveSettings()
         {
-            Pages.Children.Clear();
-            NavPanel.SelectionChanged += NavPanelOnSelectionChanged;
-            NavPanel.SelectedItem = NavPanel.MenuItems.ElementAt(0);
-        }
+            InputPage.ControllerSettings?.SaveCurrentProfile();
 
-        private void Button_Checked(object sender, RoutedEventArgs e)
-        {
-            if (sender is ToggleButton button)
+            if (Owner is MainWindow window && ViewModel.DirectoryChanged)
             {
-                if (_currentAssigner != null && button == _currentAssigner.ToggledButton)
-                {
-                    return;
-                }
-
-                if (_currentAssigner == null && (bool)button.IsChecked)
-                {
-                    _currentAssigner = new ButtonKeyAssigner(button);
-
-                    FocusManager.Instance.Focus(this, NavigationMethod.Pointer);
-
-                    PointerPressed += MouseClick;
-
-                    IKeyboard       keyboard = (IKeyboard)ViewModel.AvaloniaKeyboardDriver.GetGamepad(ViewModel.AvaloniaKeyboardDriver.GamepadsIds[0]);
-                    IButtonAssigner assigner = new KeyboardKeyAssigner(keyboard);
-
-                    _currentAssigner.GetInputAndAssign(assigner);
-                }
-                else
-                {
-                    if (_currentAssigner != null)
-                    {
-                        ToggleButton oldButton = _currentAssigner.ToggledButton;
-
-                        _currentAssigner.Cancel();
-                        _currentAssigner = null;
-
-                        button.IsChecked = false;
-                    }
-                }
+                window.ViewModel.LoadApplications();
             }
         }
 
-        private void Button_Unchecked(object sender, RoutedEventArgs e)
-        {
-            _currentAssigner?.Cancel();
-            _currentAssigner = null;
-        }
-
-        private void MouseClick(object sender, PointerPressedEventArgs e)
+        private void Load()
         {
-            bool shouldUnbind = false;
-
-            if (e.GetCurrentPoint(this).Properties.IsMiddleButtonPressed)
-            {
-                shouldUnbind = true;
-            }
-
-            _currentAssigner?.Cancel(shouldUnbind);
-
-            PointerPressed -= MouseClick;
+            Pages.Children.Clear();
+            NavPanel.SelectionChanged += NavPanelOnSelectionChanged;
+            NavPanel.SelectedItem = NavPanel.MenuItems.ElementAt(0);
         }
 
         private void NavPanelOnSelectionChanged(object sender, NavigationViewSelectionChangedEventArgs e)
         {
             if (e.SelectedItem is NavigationViewItem navitem)
```

Comment:
```suggestion
            if (e.SelectedItem is NavigationViewItem navItem && navItem.Tag is not null)
```
Change the name to `navItem` and since `navItem.Tag` is nullable add a null check.